accounts/tests/test_user.py

```diff
@@ -178,73 +180,209 @@ def test_user_delete_make_invalid_password(self):
+        user_sounds = Sound.objects.filter(user=user)
+        user_sound_ids = [s.id for s in user_sounds]
+        user.profile.delete_user(remove_sounds=True, delete_user_object_from_db=True)
+
+        self.assertFalse(User.objects.filter(id=user.id).exists())
+        self.assertTrue(DeletedUser.objects.filter(user_id=user.id).exists())
+        self.assertFalse(Comment.objects.filter(user__id=user.id).exists())
+        self.assertFalse(Thread.objects.filter(author__id=user.id).exists())
+        self.assertFalse(Post.objects.filter(author__id=user.id).exists())
+        self.assertFalse(Pack.objects.filter(user__id=user.id).exists())
+        self.assertFalse(Sound.objects.filter(user__id=user.id).exists())
+        self.assertFalse(DeletedSound.objects.filter(user__id=user.id).exists())
+        self.assertFalse(OldUsername.objects.filter(user__id=user.id).exists())
+
+        calls = [mock.call(i) for i in user_sound_ids]
+        delete_sound_from_solr.assert_has_calls(calls, any_order=True)
+        delete_sound_from_gaia.assert_has_calls(calls, any_order=True)
+
+    @mock.patch('sounds.models.delete_sound_from_gaia')
+    @mock.patch('sounds.models.delete_sound_from_solr')
+    def test_user_full_delete(self, delete_sound_from_solr, delete_sound_from_gaia):
+        # This should delete all user content, including the User object and without creating DeletedUser nor
+        # DeletedSound objects.
+        user = self.create_user_and_content()
+        user_sounds = Sound.objects.filter(user=user)
+        user_sound_ids = [s.id for s in user_sounds]
+        user.delete()
+
+        self.assertFalse(User.objects.filter(id=user.id).exists())
+        self.assertFalse(OldUsername.objects.filter(user__id=user.id).exists())
+        self.assertFalse(DeletedUser.objects.filter(user_id=user.id).exists())
+        self.assertFalse(Comment.objects.filter(user__id=user.id).exists())
+        self.assertFalse(Thread.objects.filter(author__id=user.id).exists())
+        self.assertFalse(Post.objects.filter(author__id=user.id).exists())
+        self.assertFalse(Pack.objects.filter(user__id=user.id).exists())
+        self.assertFalse(Sound.objects.filter(user__id=user.id).exists())
+        self.assertFalse(DeletedSound.objects.filter(user__id=user.id).exists())
+
+        calls = [mock.call(i) for i in user_sound_ids]
+        delete_sound_from_solr.assert_has_calls(calls, any_order=True)
+        delete_sound_from_gaia.assert_has_calls(calls, any_order=True)
+
+    def test_user_delete_include_sounds_using_web_form(self):
+        # Test user's option to delete user account including the sounds
 
-    def test_user_delete_using_form(self):
-        # This should set user's attribute active to false and anonymize it
         user = self.create_user_and_content(is_index_dirty=False)
-        a = self.client.login(username=user.username, password='testpass')
+        self.client.login(username=user.username, password='testpass')
```

Comment:
can use force_login here to not need a password

Reply:
Refactored all fs tests to use `force_login`. This is something I wanted to do some time ago... e153bcbefcd1fd1de30d11f65cb2c4b7082eb515